File: StudentCard.Application/Students/Dtos/PersonLotDtos/BaseDtos/BaseSemesterDto.cs
using StudentCard.Infrastructure.Helpers.Dtos;

namespace StudentCard.Application.Students.Dtos.PersonLotDtos.BaseDtos
{
    public class BaseSemesterDto
    {
        public int Id { get; set; }
        public StudentStatusDto StudentStatus { get; set; }
        public NomenclatureDto StudentEvent { get; set; }

        public NomenclatureDto EducationFeeType { get; set; }
    }
}

File: WebApplication/src/modules/student/dtos/person-lot.dtos/base.dtos/base-semester.dto.ts
import { NomenclatureDto } from "src/infrastructure/models/nomenclature.dto";
import { StudentStatusDto } from "../helpers/student-status.dto";

export class BaseSemesterDto {
  studentStatus: StudentStatusDto;
  studentEvent: NomenclatureDto;
  educationFeeType: NomenclatureDto;
}